File: src/main/java/net/guizhanss/fastmachines/implementation/items/machines/infinityexpansion/FastInfinityWorkbench.java
package net.guizhanss.fastmachines.implementation.items.machines.infinityexpansion;

import org.bukkit.Material;
import org.bukkit.inventory.ItemStack;

import io.github.mooy1.infinityexpansion.items.blocks.InfinityWorkbench;
import io.github.thebusybiscuit.slimefun5.api.items.ItemGroup;
import io.github.thebusybiscuit.slimefun5.api.items.SlimefunItemStack;
import io.github.thebusybiscuit.slimefun5.api.recipes.RecipeType;
import io.github.thebusybiscuit.slimefun5.libraries.xseries.XMaterial;

import net.guizhanss.fastmachines.FastMachines;
import net.guizhanss.fastmachines.core.recipes.loaders.InfinityExpansionRecipeLoader;
import net.guizhanss.fastmachines.core.recipes.loaders.RecipeLoader;
import net.guizhanss.fastmachines.implementation.items.machines.base.BaseFastMachine;
import net.guizhanss.fastmachines.utils.MaterialCompat;

public class FastInfinityWorkbench extends BaseFastMachine {

    public FastInfinityWorkbench(ItemGroup itemGroup, SlimefunItemStack itemStack, RecipeType recipeType, ItemStack[] recipe) {
        super(itemGroup, itemStack, recipeType, recipe, 100_000_000, 10_000_000);
    }

    @Override
    public Material getCraftItemMaterial() {
        return MaterialCompat.safe(XMaterial.RESPAWN_ANCHOR);
    }

    @Override
    public RecipeLoader getRecipeLoader() {
        return new InfinityExpansionRecipeLoader(this, InfinityWorkbench.class);
    }

    @Override
    public boolean registerPrecondition() {
        return FastMachines.getIntegrationService().isInfinityExpansionEnabled();
    }
}

File: src/main/kotlin/net/guizhanss/fastmachines/implementation/items/machines/infinityexpansion/FastInfinityWorkbench.kt
package net.guizhanss.fastmachines.implementation.items.machines.infinityexpansion

import io.github.mooy1.infinityexpansion.items.blocks.InfinityWorkbench
import io.github.thebusybiscuit.slimefun5.api.items.ItemGroup
import io.github.thebusybiscuit.slimefun5.api.items.SlimefunItemStack
import io.github.thebusybiscuit.slimefun5.api.recipes.RecipeType
import io.github.thebusybiscuit.slimefun5.libraries.xseries.XMaterial
import net.guizhanss.fastmachines.utils.MaterialCompat
import net.guizhanss.fastmachines.FastMachines
import net.guizhanss.fastmachines.core.recipes.loaders.InfinityExpansionRecipeLoader
import net.guizhanss.fastmachines.core.recipes.loaders.RecipeLoader
import net.guizhanss.fastmachines.implementation.items.machines.base.BaseFastMachine
import org.bukkit.Material
import org.bukkit.inventory.ItemStack

class FastInfinityWorkbench(
    itemGroup: ItemGroup,
    itemStack: SlimefunItemStack,
    recipeType: RecipeType,
    recipe: Array<out ItemStack?>,
) : BaseFastMachine(itemGroup, itemStack, recipeType, recipe, 100_000_000, 10_000_000) {

    override val craftItemMaterial: Material
        get() = MaterialCompat.safe(XMaterial.RESPAWN_ANCHOR)

    override val recipeLoader: RecipeLoader
        get() = InfinityExpansionRecipeLoader(this, InfinityWorkbench::class.java)

    override fun registerPrecondition() = FastMachines.integrationService.infinityExpansionEnabled
}
